internal/controller/volumereplicationgroup_controller.go

```diff
@@ -966,6 +916,14 @@ func (v *VRGInstance) separatePVCUsingPeerClassAndSC(peerClasses []ramendrv1alph
 		if peerClass.ReplicationID != "" {
 			replicationClass := v.findReplicationClassUsingPeerClass(peerClass, storageClass)
 			if replicationClass != nil {
+				// label VolRep PVCs if peerClass.grouping is enabled
+				if peerClass.Grouping {
+					if err := v.addVolRepConsistencyGroupLabel(pvc); err != nil {
```

Comment:
I am wondering if we should add the CG label in `preparePVCForVRProtection` instead of here. All PVC updates happen in that function, and it is invoked before checking if the PVC needs group protection or not.

The parallel for volsync would/could be here: `reconcilePVCAsVolSyncPrimary`

@BenamarMk thoughts?